Changes to the Readme. (#4)
ChartIQ/finsemble-fdc3: README.md

3) Clone this repo
   - **our advice:** clone this repo to the same directory as the seed-project e.g *myfolder/finsemble-seed* & *myfolder/finsemble-fdc3*
4) If you clone in a different location, open **finsemble.config.json** and update `seedProjectDirectory` with the path to your local Finsemble Seed Project.
5) Inside this project run `npm install` then run `npm run watch` 
**Note:** The watch script will continue to monitor this project for file changes. If you do not need it to, it  can be stopped once all the files have been copied to the seed project (which takes approx. 30 seconds).
6) Follow the instructions on how to add the FDC3 to your [Component](###-FDC3-in-Finsemble-Components) / [Service](###FDC3-in-Finsemble-Services):

## Using FDC3 in Finsemble
The Finsemble implementation of the FDC3 DesktopAgent and Channels APIs can be used in both Finsemble Components and Services.

### FDC3 in Finsemble Components
The FDC3 client is most easily made available to your components via preload, which will give your component access to it as if it were any other Finsemble client. It is anticipated that later versions of Finsemble will include the FDC3 client by default.

> **Component Config**

Reviewer: ```suggestion
#### Component Config
```
Ended up running into next line, better stick to heading styles

To use the FDC3 client in your component, you will need to make 3 additions to your component's configuration: 
- Preload the client, 
- Ensure that you have a Toolbar Icon URL set 
- Add FDC3 specific configuration

> **Preload**

Reviewer: ```suggestion
##### Preload
```

To add the FDC3 preload to your component, set `component.preload` in its configuration to the path to the built preload script, e.g. `"$applicationRoot/preloads/FDC3Client.js"`, or if using multiple preload scripts add it to an array of their paths (see the config example below).

> **Toolbar Icon URL**

Reviewer: ```suggestion
##### Toolbar Icon URL
```

The icon url is used by the intent resolver to display the logo of your component. This URL can use the `$applicationRoot` variable to set a local path or can be an external URL. Supported formats: `.jpg`, `.png`, `.svg`.

> **FDC3 configuration**

Reviewer: ```suggestion
##### FDC3 configuration
```

6) Follow the instruction on how to add the FDC3 to your [Component](###-FDC3-in-Finsemble-Components) / [Service](###FDC3-in-Finsemble-Services):

## Finsemble FDC3 Implementation

### FDC3 in Finsemble Components
_The FDC3 Client is added in via preload, you now have access to this client as if it were any other Finsemble client._

#### Component Config:
The Finsemble config file allows Finsemble to understand that your application is built with FDC3 capability. You will need to add three sections to your component config to start working with FDC3 in Finsemble - Preload, Toolbar Icon URL and FDC3.

<a href="preload" title="#preload">**Preload:**</a>
The FDC3 preload is required to enable the FDC3 Client in your components to use the FDC3 Client API. Adding the preload is as simple as adding the preload to the preload section (see the config example below).

**Toolbar Icon URL:**
The icon url is used by the intent resolver to display the logo of the your component. This URL can use the $applicationRoot or can be an external URL. Supported formats: jpg, png, svg

**FDC3:**

> This section includes the intents and context that your

Reviewer: ```suggestion
An additional section in your component config, `foreign.services.fdc3`, is used to specify intents and associated context types that your
```
probably should also remove the line break here

```

### FDC3 in Finsemble Services

> Please see our [exampleFDC3Service](./src/services/testFDC3) for how to structure your service to work with FDC3.

Reviewer: ```suggestion
The Finsemble FDC3 client may also be used in Finsemble services, where it is imported into the service implementation. Please see our [exampleFDC3Service](./src/services/testFDC3) for an example of how to structure your service to work with FDC3.
```

## API

> 

Reviewer: ```suggestion
Please note that the FINOS FDC3 website provides documentation for the [Desktop Agent](https://fdc3.finos.org/docs/1.1/api/ref/DesktopAgent) and [Channel](https://fdc3.finos.org/docs/1.1/api/ref/Channel) APIs. The main API calls are also detailed here for convenience.
```

> ### App:

Reviewer: ```suggestion
### App
```

<br/>

> ### Context:

Reviewer: ```suggestion
### Context
```

<br/>

> ### Intents:

Reviewer: ```suggestion
### Intents
```

<br/>

> ### Channels:

Reviewer: ```suggestion
### Channels
```

> Example of the FDC3 Channel Interface:

Reviewer: ```suggestion
The DesktopAgent API will allow you to join, listen to and broadcast context to a single 'channel' via its API. However, you can also interact directly with FDC3 channels, which have the following interface:
```

> ```

Reviewer: ```suggestion
```javascript
```

## Example use cases:

> When a user clicks on the chart button I want to open my ChartIQ Chart, how would I do this?

There is a simple command for this! `fdc3.open("ChartIQ")`

> My Chart opens but now I need to send instrument data to it, what do I need to do next?

The fdc3.open call allows you to send context with it too.
Here is an example of opening the chart and sending the GOOGL (Google) ticker symbol to it.

> ```

Reviewer: ```suggestion
```javascript
```

 });
```

> I want to send an instrument (MSFT) but I want the user to choose the application they want to display it with, how would do I that?

This is where you can raise an intent. Once the intent has been raised it will show all the apps capable of dealing with your intent in the intent resolver. This works in a similar fashion to your phone when you want to share a link or open a calendar invite and it asks which application you would like to use.
To do this you need two things
1) A Finsemble component that has been set up to accept your intent (you can read the section on config above to do this).
2) Your component to have a button or event to run the following code

> ```

Reviewer: ```suggestion
```javascript
```

> Is there a way I can find out intents in advance so that if there are no applications that accept my intent I can do something else with a user request?
>
Yes. You have two options to find intents, you can
`fdc3.findIntent("ViewChart", context); //context object is optional`
OR
`fdc3.findIntentsByContext(context);`

> I don't want to use the intent resolver each time I want to send context (data) between components, is there another way?
>
FDC3 1.1 introduces the Channels API. You can connect your components to a channel either programmatically or user-led via the Finsemble Linker.
Channels then allow you to share context between them. A single application (component) can only join one channel at a time e.g it cannot join the purple and the yellow channel, if the user tries to do this they will be removed from the last channel joined. You can however programmatically listen to as many channels as you like from your application.
Many applications can be joined to a single-channel e.g App A, B and C can all be joined by the purple channel.

> What do I need to do to enable the channels so that my end users can start using it?
>

> Finsemble allows you to use the Linker Channels. All you need to do is add the FDC3 preload to your component then set up a [context listener](#context-listener) and Finsemble will handle the rest. Your end-users can now use the Finsemble Linker button to change the FDC3 channel they are joined to.

Reviewer: ```suggestion
Finsemble includes user interface for selecting built-in (or system) colour channels, normally used with the Finsemble Linker API. The FInsemble FDC3 implementation, if preloaded into a component, will take over the Linker UI allowing you to use it select FDC3 system channels instead. All you need to do is add the FDC3 preload to your component then set up a [context listener](#context-listener) and Finsemble will handle the rest. Your end-users can now use the Finsemble Linker button to change the FDC3 channel they are joined to.
```

@WatsonCIQ can you confirm if they also need to enable the linker in their config or whether the preload does this for you please.

> How do I know what channels I have access to?
>

> System channels (including a global channel) are listed by doing the following:

Reviewer: ```suggestion
You can list the available System channels (including a global channel) by doing the following:
```

`const systemChannels = await fdc3.getSystemChannels();
// Array of System Channels`

> Am I limited to just your Finsemble Linker channels?
>
No. You can add as many channels as you like, try it by using `const Channel = fdc3.getOrCreateChannel("myChannel")`

> You mentioned programmatically sending context (data) over channels how do I do that?
>
If you want to send context via the channel that you are currently joined to you can do the following:

> ```

Reviewer: ```suggestion
```javascript
```

const instrument = {
  type: 'fdc3.instrument',
  id: { ticker: 'MSFT' }
};
fdc3.broadcast(instrument);
```

If you want to send context to a channel you are not joined to you need to get the channel first.

> ```

Reviewer: ```suggestion
```javascript
```

try {
  const myChannel = await fdc3.getOrCreateChannel("myChannel");

  const instrument = {
    type: 'fdc3.instrument',
    id: { ticker: 'MSFT' }
    };

  const myChannel.broadcast(instrument);

} catch (err){
  //app could not register the channel
};
```

> Now I have sent context (data) how do I listen for it?

It depends on if you want to listen for context on the channel you are joined to or a different channel(s).

If you are joined to a channel you can use the following:
```
// any context
const listener = fdc3.addContextListener(context => { ... });

// listener for a specific type
const contactListener = fdc3.addContextListener('fdc3.contact', contact => { ... });
```

If you want to listen to a specific channel you need to get it first like this:

> ```

Reviewer: ```suggestion
```javascript
```

> Is there a way I can find out intents in advance so that if there are no applications that accept my intent I can do something else with a user request?
>
Yes. You have two options to find intents, you can
`fdc3.findIntent("ViewChart", context); //context object is optional`
OR
`fdc3.findIntentsByContext(context);`

> I don't want to use the intent resolver each time I want to send context (data) between components, is there another way?
>
FDC3 1.1 introduces the Channels API. You can connect your components to a channel either programmatically or user-led via the Finsemble Linker.
Channels then allow you to share context between them. A single application (component) can only join one channel at a time e.g it cannot join the purple and the yellow channel, if the user tries to do this they will be removed from the last channel joined. You can however programmatically listen to as many channels as you like from your application.
Many applications can be joined to a single-channel e.g App A, B and C can all be joined by the purple channel.

> What do I need to do to enable the channels so that my end users can start using it?
>

> Finsemble includes user interface for selecting built-in (or system) colour channels, normally used with the Finsemble Linker API. The FInsemble FDC3 implementation, if preloaded into a component, will take over the Linker UI allowing you to use it select FDC3 system channels instead. All you need to do is add the FDC3 preload to your component then set up a [context listener](#context-listener) and Finsemble will handle the rest. Your end-users can now use the Finsemble Linker button to change the FDC3 channel they are joined to.

Reviewer 2: ```suggestion
Finsemble includes user interface for selecting built-in (or system) color channels, normally used with the Finsemble Linker API. The Finsemble FDC3 implementation, if preloaded into a component, will take over the Linker UI allowing you to use it select FDC3 system channels instead. All you need to do is add the FDC3 preload to your component then set up a [context listener](#context-listener) and Finsemble will handle the rest. Your end-users can now use the Finsemble Linker button to change the FDC3 channel they are joined to.
```

`const systemChannels = await fdc3.getSystemChannels();
// Array of System Channels`

> Am I limited to just your Finsemble Linker channels?
>
No. You can add as many channels as you like, try it by using `const Channel = fdc3.getOrCreateChannel("myChannel")`

> You mentioned programmatically sending context (data) over channels how do I do that?
>
If you want to send context via the channel that you are currently joined to you can do the following:
```javascript
const instrument = {
  type: 'fdc3.instrument',
  id: { ticker: 'MSFT' }
};
fdc3.broadcast(instrument);
```

If you want to send context to a channel you are not joined to you need to get the channel first.
```javascript
try {
  const myChannel = await fdc3.getOrCreateChannel("myChannel");

  const instrument = {
    type: 'fdc3.instrument',
    id: { ticker: 'MSFT' }
    };

  const myChannel.broadcast(instrument);

} catch (err){
  //app could not register the channel
};
```

> Now I have sent context (data) how do I listen for it?

It depends on if you want to listen for context on the channel you are joined to or a different channel(s).

If you are joined to a channel you can use the following:

> ```

Author: ```suggestion
```javascript
```
